packages/passport-interface/src/api/makeRequest.ts

```diff
@@ -0,0 +1,119 @@
+import { getErrorMessage } from "@pcd/util";
+import urlJoin from "url-join";
+import { APIResult, GetResultValue, ResultMapper } from "./apiResult";
+import { POST } from "./constants";
+
+/**
+ * Wrapper of {@link httpRequest} that sends a GET request.
+ */
+export async function httpGet<T extends APIResult<unknown, unknown>>(
+  url: string,
+  opts: ResultMapper<T>,
+  query?: object
+): Promise<T> {
+  return httpRequest<T>(
+    urlJoin(
+      url,
+      "?" + new URLSearchParams((query as Record<string, string>) ?? {})
+    ),
+    opts
+  );
+}
+
+/**
+ * Wrapper of {@link httpRequest} that sends a POST request.
+ */
+export async function httpPost<T extends APIResult<unknown, unknown>>(
+  url: string,
+  opts: ResultMapper<T>,
+  postBody?: object
+): Promise<T> {
+  return httpRequest<T>(url, opts, postBody);
+}
+
+/**
+ * Shorthand for a {@link httpGet} whose error type is a string.
+ */
+export async function httpGetSimple<TResult>(
+  url: string,
+  onValue: GetResultValue<APIResult<TResult, string>>,
+  query?: object
+): Promise<APIResult<TResult, string>> {
```

Comment:
for me, it's a bit unintuitive to have the `onValue` param in between `url` and `query`. it does make sense to keep `query` as an optional param, so one option could be something like this


```typescript
export async function httpGetSimple<TResult>(
  onValue: GetResultValue<APIResult<TResult, string>>,
  url: string,
  query?: object
): Promise<APIResult<TResult, string>> {
```

or even this

```typescript
export async function httpGetSimple<TResult>(
  url: string,
  options: { query?: object; onValue?: GetResultValue<APIResult<TResult, string>> }
): Promise<APIResult<TResult, string>> {
```